### PR TITLE
TAO-10365 Added new interface implementation to PermissionProvider

### PR DESCRIPTION
**Pull request by luisfn**
Added interface/method to be able to retrieve access data from resources

### CODE REVIEW

File: model/PermissionProvider.php [hunk 1]
```diff
@@ -136,4 +137,9 @@ public static function getSupportedRootClasses()
             new core_kernel_classes_Class(GenerisRdf::CLASS_ROLE)
         ];
     }
+
+    public function getResourceAccessData(core_kernel_classes_Resource $resource): array
+    {
+        return AdminService::getUsersPermissions($resource->getUri());
```
Comment (jbout) [inline model/PermissionProvider.php:143]: Good improvement, but moving the functionality to getResourceAccessData() and referencing it from AdminService::getUsersPermissions() for backward compatibility would be cleaner (as it removes the `ServiceManager::getServiceManager()` from the chain)

Comment (luisfn) [reply model/PermissionProvider.php]: Tried to move things and added a test for the new method. Please have a look

File: model/PermissionProvider.php [hunk 2]
```diff
@@ -136,4 +137,23 @@ public static function getSupportedRootClasses()
...
+        $db = $this->getServiceLocator()->get(DataBaseAccess::SERVICE_ID);
+        $results = $db->getUsersWithPermissions([$resourceId]);
+
+        $permissions = [];
+        foreach ($results as $result) {
+            $user = $result['user_id'];
+
+            if (!isset($permissions[$user])) {
+                $permissions[$user] = [];
+            }
```
Comment (wazelin) [inline model/PermissionProvider.php:153]:
```suggestion
```

It's not a problem to create a new array entry and add a new element into it at the same time.

File: test/unit/model/PermissionProviderTest.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        $result = $this->sut->getResourceAccessData('id');
+        $expected = [
+            'http://www.tao.lu/Ontologies/TAO.rdf#BackOfficeRole' => ['READ', 'WRITE', 'GRANT']
+        ];
+
+
+        self::assertSame($expected, $result);
+    }
+
+    private function mockDatabaseAccess()
```
Comment (wazelin) [inline test/unit/model/PermissionProviderTest.php:56]:
```suggestion
        $databaseAccess = $this->createDatabaseAccessMock();
        $serviceLocator = $this->getServiceLocatorMock([
            DataBaseAccess::SERVICE_ID => $databaseAccess
        ]);

        $this->sut = new PermissionProvider();
        $this->sut->setServiceLocator($serviceLocator);
    }

    public function testGetResourceAccessData(): void
    {
        $result = $this->sut->getResourceAccessData('id');
        $expected = [
            'http://www.tao.lu/Ontologies/TAO.rdf#BackOfficeRole' => ['READ', 'WRITE', 'GRANT']
        ];

        self::assertSame($expected, $result);
    }

    private function createDatabaseAccessMock(): DataBaseAccess
```

File: manifest.php [hunk 4]
```diff
@@ -34,7 +34,7 @@
     'author' => 'Open Assessment Technologies SA',
     'requires' => [
        'taoBackOffice' => '>=3.0.0',
-       'generis' => '>=12.15.0',
+       'generis' => 'dev-feature/TAO-10365-apply-access-control-to-queries',
```
Comment (wazelin) [inline manifest.php:37]:
```suggestion
       'generis' => '>=13.1.0',
```

**Comment by wazelin** [comment]
> The build is failing, could you please check?

It's gonna be fine once `generis`' PR is merged.

**Pull request merged**